### ISSUE TITLE
Not working with some types

### ISSUE THREAD

**Issue by lishine** [theKashey/memoize-state#21]
Wrapping with memoize redux selectors and passing them to connect.
When selector returns state or state.location (an object that prints in the console like proxy), an error happens.

```
state.location Proxy {id: "Login", params: {…}, query: {…}, prev: null}
00:04:02.369 VM47658 connectAdvanced.js:251 Uncaught TypeError: result.hasOwnProperty is not a function
    at deproxifyResult (VM47694 call.js:65)
    at deproxifyResult (VM47694 call.js:72)
    at deproxifyResult (VM47694 call.js:72)
    at deproxifyResult (VM47694 call.js:48)
    at deproxifyResult (VM47694 call.js:72)
    at callIn (VM47694 call.js:141)
    at Function.ms_ [as mapToProps] (VM47686 memoize.js:82)
    at mapToPropsProxy (VM47669 wrapMapToProps.js:48)
    at Function.detectFactoryAndVerify (VM47669 wrapMapToProps.js:57)
    at mapToPropsProxy (VM47669 wrapMapToProps.js:48)
deproxifyResult @ VM47694 call.js:65
deproxifyResult @ VM47694 call.js:72
deproxifyResult @ VM47694 call.js:72
deproxifyResult @ VM47694 call.js:48
deproxifyResult @ VM47694 call.js:72
callIn @ VM47694 call.js:141
ms_ @ VM47686 memoize.js:82
mapToPropsProxy @ VM47669 wrapMapToProps.js:48
detectFactoryAndVerify @ VM47669 wrapMapToProps.js:57
mapToPropsProxy @ VM47669 wrapMapToProps.js:48
handleFirstCall @ VM47683 selectorFactory.js:31
pureFinalPropsSelector @ VM47683 selectorFactory.js:79
runComponentSelector @ VM47658 connectAdvanced.js:36
initSelector @ VM47658 connectAdvanced.js:188
Connect(Component) @ VM47658 connectAdvanced.js:129
constructClassInstance @ VM47646 react-dom.development.js:11768
updateClassComponent @ VM47646 react-dom.development.js:13490
beginWork @ VM47646 react-dom.development.js:14089
performUnitOfWork @ VM47646 react-dom.development.js:16415
workLoop @ VM47646 react-dom.development.js:16453
callCallback @ VM47646 react-dom.development.js:145
invokeGuardedCallbackDev @ VM47646 react-dom.development.js:195
invokeGuardedCallback @ VM47646 react-dom.development.js:248
replayUnitOfWork @ VM47646 react-dom.development.js:15744
renderRoot @ VM47646 react-dom.development.js:16547
performWorkOnRoot @ VM47646 react-dom.development.js:17386
performWork @ VM47646 react-dom.development.js:17294
performSyncWork @ VM47646 react-dom.development.js:17266
requestWork @ VM47646 react-dom.development.js:17154
scheduleWork @ VM47646 react-dom.development.js:16948
scheduleRootUpdate @ VM47646 react-dom.development.js:17636
updateContainerAtExpirationTime @ VM47646 react-dom.development.js:17663
updateContainer @ VM47646 react-dom.development.js:17690
ReactRoot.render @ VM47646 react-dom.development.js:17956
(anonymous) @ VM47646 react-dom.development.js:18096
unbatchedUpdates @ VM47646 react-dom.development.js:17517
legacyRenderSubtreeIntoContainer @ VM47646 react-dom.development.js:18092
render @ VM47646 react-dom.development.js:18151
(anonymous) @ VM47639 index.js:15
(anonymous) @ VM47639 index.js:23
./src/client/index.js @ VM47334 bundle.js:11222
__webpack_require__ @ VM47334 bundle.js:725
fn @ VM47334 bundle.js:102
0 @ VM47334 bundle.js:11668
__webpack_require__ @ VM47334 bundle.js:725
(anonymous) @ VM47334 bundle.js:792
(anonymous) @ VM47334 bundle.js:795
00:04:02.389 VM47646 react-dom.development.js:14549 The above error occurred in the <Connect(Component)> component:
    in Connect(Component)
    in div
    in Unknown (created by ReactiveHOC)
    in div (created by Reflex)

```

```
import memoize from 'memoize-state'
import { connect as rrConnect } from 'react-redux'

const mem = o =>
	memoize((s, p) =>
		Object.entries(o).reduce((acc, [key, func]) => Object.assign(acc, { [key]: func(s, p) }), {})
	)

export const connect = selectors => rrConnect(mem(selectors))

```

**Comment by theKashey** [comment]
I know “why” this error could happen, and know the way to solve it. 
Anyway - how do you actually got into it? Could you share “selectors”, and the “state” - I need to figure out how did you created an object without prototype.

**Comment by lishine** [comment]
https://github.com/lishine/login-full-stack/blob/master/src/client/store/configureStore.js
https://github.com/lishine/login-full-stack/blob/master/src/client/store/rootReducer.js
It helps?
It happens if a selector returns an object: state or state.location
export const getState = state => state

**Comment by theKashey** [comment]
Sounds reasonable. Fix is simply, but I’ll release an update only when got off aircraft:)

**Comment by theKashey** [comment]
Fixed in 2.0.4
